### DISCUSSION TITLE
Suggested change to module compilation

### DISCUSSION THREAD

**Discussion by cpurdy** [xtclang/xvm#33] [Ideas]
In the Ecstasy Guide ([https://github.com/xtclang/xvm/wiki/lang-module](https://github.com/xtclang/xvm/wiki/lang-module)), we talk about how no code can exist outside of the module statement, including the obvious "import" statements that Java developers are so used to. While it is not an arbitrary decision (there are real technical reasons for the rule), it seems like an easy "gotcha" to fix, by simply relocating import statements within the same file from outside of the module to inside of the module.

Thoughts?

**Comment by ggleyzer** [comment]
I like that idea... After all, we do allow comments to be placed outside of the module [curlies], why not make an exceptions for imports as well

**Comment by cpurdy** [comment]
The compiler change was made, and I updated the wiki docs a few days ago to reflect this change.
